heat/classification/tests/test_knn.py

```diff
@@ -11,14 +11,9 @@ def test_split_none(self):
         x = ht.load_hdf5("heat/datasets/iris.h5", dataset="data")
 
         # generate keys for the iris.h5 dataset
-        keys = []
-        for i in range(50):
-            keys.append(0)
-        for i in range(50, 100):
-            keys.append(1)
-        for i in range(100, 150):
-            keys.append(2)
-        y = ht.array(keys)
+        y = ht.zeros(150, dtype=ht.int32)
+        y[50:100] = 1
```

Comment:
Can you undo your changes only in this file in order to test that this is the location of the error, please?

Reply:
@Sai-Suraj-27 I don't think you need to undo all the changes, the error message indicates that arrays used for indexing must be double precision

Reply:
ok I am sorry, didnt know how to interpret the error correctly... so I guess Claudia is right and what I suggested is nonsense.. @Sai-Suraj-27  sry for that

Reply:
Sorry, mam. I did it, but I still have all the old code (you wrote yesterday), Should I again push that?

Reply:
I would suggest the following: I run the CI on the old version in order to find out whether the reason for the failure was really located in the test_knn-file. If this is the case, we can correct the error as Claudia proposed.

Reply:
ok, it looks like that indeed these changes caused (for whatever reason onyl in certain constellations) the problems... so we can use Claudias hint to solve them

Reply:
@mrfh92 sir, I think changing the `dtype=ht.int32` to `dtype=ht.int64` while declaring `y` like this: `y = ht.zeros(150, dtype=ht.int64)` might solve the error, because it shows in the error that "IndexError: tensors used as indices must be long, byte or bool tensors", and `int64` corresponds to a "long" type in Python. Shall we try this? I will make the required changes and push the code. If you are ok with trying the `dtype=ht.int64`. Thank you.